src/main/java/ch/njol/skript/expressions/ExprPotionEffect.java

```diff
@@ -36,21 +37,29 @@
 import ch.njol.util.Kleenean;
 
 @Name("Potion Effect")
-@Description({"Create a new potion effect to apply to an entity or item type. Do note that when applying potion effects ",
-	"to tipped arrows/lingering potions, Minecraft reduces the timespan."})
-@Examples({"set {_p} to potion effect of speed of tier 1 without particles for 10 minutes",
-	"add {_p} to potion effects of player's tool",
-	"add {_p} to potion effects of target entity",
-	"add potion effect of speed 1 to potion effects of player"})
-@Since("2.5.2")
+@Description({
+		"Create a new potion effect to apply to an entity or item type. Do note that when applying potion effects ",
+		"to tipped arrows/lingering potions, Minecraft reduces the timespan."
+})
+@Examples({
+		"set {_p} to potion effect of speed of tier 1 without particles for 10 minutes",
+		"add {_p} to potion effects of player's tool",
+		"add {_p} to potion effects of target entity",
+		"add potion effect of speed 1 to potion effects of player"
+})
+@Since("2.5.2, INSERT VERSION (new syntax)")
 public class ExprPotionEffect extends SimpleExpression<PotionEffect> {
+
 	static {
 		Skript.registerExpression(ExprPotionEffect.class, PotionEffect.class, ExpressionType.COMBINED,
-			"[new] potion effect of %potioneffecttype% [potion] [[[of] tier] %-number%] [(1¦without particles)] [for %-timespan%]",
-			"[new] ambient potion effect of %potioneffecttype% [potion] [[[of] tier] %-number%] [(1¦without particles)] [for %-timespan%]");
+				"[new] potion [effect] of %potioneffecttype% [potion] [[[of] tier] %-number%] [(1¦without particles)] [for %-timespan%]",
+				"[new] [potion [effect] of] %potioneffecttype% [potion] [[of] tier] %number% [(1¦without particles)] [for %-timespan%]", // "speed 2"
+				"[new] ambient potion [effect] of %potioneffecttype% [potion] [[[of] tier] %-number%] [(1¦without particles)] [for %-timespan%]",
+				"[new] ambient [potion [effect] of] %potioneffecttype% [potion] [[of] tier] %number% [(1¦without particles)] [for %-timespan%]" // "ambient speed 2"
+		);
```

Comment:
the issue with these patterns is that they allow stuff such as:
```
potion of speed potion
potion of speed tier {_a} potion
swiftness {_r}
ambient potion of swiftness tier {_r} potion for {_z}
```
I'd rather see something like this instead:
```
[new] [(1¦ambient)] %potioneffect% potion [(1¦without particles)] [for %-timespan%]
[new] [(1¦ambient)] potion of %potioneffect% [(1¦without particles)] [for %-timespan%]
```
the potioneffect classinfo would allow literals such as `speed`, `speed 2`, `speed of tier 3`. This would require users to use the parse expression to do things such as `speed {_r}` but it'd make the intent a bit more clear in my opinion

Reply:
I suppose I could try and simplify the patterns here. These patterns aren't actually made by me, they are just the patterns that already existed when this effect was fully implemented previously. The two new ones are just to allow something like "speed 2" without having a pattern that is just "%potioneffecttype%"

If I am already making some breaking changes though, maybe it is worth it to go the whole way. I'm not sure if I'm in favor of breaking something like "speed {_amount}" though.